src/snapred/backend/recipe/algorithm/PixelDiffractionCalibration.py

```diff
@@ -72,15 +77,22 @@ def chopIngredients(self, ingredients: Ingredients) -> None:
         for peakList in ingredients.groupedPeakLists:
             self.maxDSpaceShifts[peakList.groupID] = 2.5 * peakList.maxfwhm
 
-        # create string name for output calibration table
-        # TODO: use workspace namer
+        # create a default name for output calibration table
         self.DIFCpixel: str = ""
         if self.getProperty("CalibrationTable").isDefault:
-            self.DIFCpixel = f"_DIFC_{self.runNumber}"
+            self.DIFCpixel = wng.diffCalTable().runNumber(self.runNumber).build()
             self.setProperty("CalibrationTable", self.DIFCpixel)
         else:
             self.DIFCpixel = self.getPropertyValue("CalibrationTable")
 
+        # create a default name for output mask workspace
```

Comment:
Same as previous comment on required outputworkspace names.  I dont think I've encountered a situation where we only need partial results from out algo's, and I would imagine we we have to refactor and split them if it is indeed required.